### PR TITLE
Connect4

### PR DESCRIPTION
**Pull request by GAIGResearch**
A classic: Connect 4. Game heuristic is a simple win/lose heuristic.

### CODE REVIEW

File: src/main/java/games/connect4/Connect4Constants.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import core.components.Token;
+
+import java.util.ArrayList;
+public class Connect4Constants {
+    public static final ArrayList<Token> playerMapping = new ArrayList<Token>() {{
+        add(new Token("x"));
+        add(new Token("o"));
+    }};
+    public static final String emptyCell = ".";
+    public static final int winCount = 4;
```
Comment (rdgain) [inline src/main/java/games/connect4/Connect4Constants.java:11]: Parameter please :)

File: src/main/java/games/connect4/Connect4GameState.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    public GridBoard<Token> getGridBoard() {
+        return gridBoard;
+    }
+
+    @Override
+    public void printToConsole() {
+        System.out.println(gridBoard.toString());
+    }
+
+    public void registerWinningCells(LinkedList<Pair<Integer, Integer>> winnerCells) {
```
Comment (rdgain) [inline src/main/java/games/connect4/Connect4GameState.java:96]: Package private?

**Review by rdgain — CHANGES_REQUESTED** [review]
A couple of small changes, the rest is approved.

File: src/main/java/games/connect4/Connect4Constants.java (hunk 1 above)
Comment (GAIGResearch) [reply src/main/java/games/connect4/Connect4Constants.java]: Done.

File: src/main/java/games/connect4/Connect4GameState.java (hunk 2 above)
Comment (GAIGResearch) [reply src/main/java/games/connect4/Connect4GameState.java]: Done.

**Review by rdgain — APPROVED** [review]

**Pull request merged**